test/multiverse/suites/roda/roda_instrumentation_test.rb

```
@@ -0,0 +1,156 @@
+
+    assert_equal ::NewRelic::Agent::UNKNOWN_METRIC, name
+  end
+
+  def test_request_is_recorded
+    get('/home')
+    txn = harvest_transaction_events![1][0]
+
+    assert_equal 'Controller/Roda/RodaTestApp/GET home', txn[0]['name']
+    assert_equal 200, txn[2][:'http.statusCode']
+  end
+
+  def test_500_response_status
+    get('/error')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events!
+
+    assert_equal 500, txn[1][0][2][:"http.statusCode"]
+    assert_equal 'Oh No!', last_response.body
+    assert_equal 1, errors.size
+  end
+
+  def test_404_response_status
+    get('/nothing')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events!
+
+    assert_equal 404, txn[1][0][2][:"http.statusCode"]
+    assert_equal 0, errors.size
+  end
+
+  def test_empty_route_name_and_response_status
+    get('')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events![1][0]
+
+    assert_equal 'Controller/Roda/RodaTestApp/GET /', txn[0]['name']
+    assert_equal 302, txn[2][:'http.statusCode']
+  end
+
+  def test_roda_auto_middleware_disabled
+    with_config(:disable_roda_auto_middleware => true) do
+      RodaNoMiddleware.build_rack_app_with_tracing {}
+
+      assert_truthy NewRelic::Agent::Agent::config[:disable_roda_auto_middleware]
+    end
+  end
+
+  def test_roda_instrumentation_works_if_middleware_disabled
+    with_config(:disable_middleware_instrumentation => true) do
+      get('/home')
+      txn = harvest_transaction_events![1][0]
+
+      assert_equal 'Controller/Roda/RodaTestApp/GET home', txn[0]['name']
+    end
+  end
+
+  def test_transaction_name_error
+    NewRelic::Agent.stub(:logger, NewRelic::Agent::MemoryLogger.new) do
+      begin
+        NewRelic::Agent::Instrumentation::Roda::TransactionNamer.transaction_name({})
+      rescue
+        # NOOP - Allow error to be raised
+      ensure
+        assert_logged(/Error encountered trying to identify Roda transaction name/)
+      end
```

You can:

- Do away with the `rescue`, as the instrumentation swallows the underlying exception and doesn't re-raise it
- Enhance your existing logger output check to confirm that the expected exception was the one that was logged
- Add an additional assertion to confirm that the return value is what's expected

We can also update `test_rack_request_params_error` in a similar fashion, but I'm waiting for the suggested change to the code being tested to be accepted/rejected first.

```suggestion
      # pass in {} to produce an error, because {} doesn't support #path and
      # confirm that the desired error handling took place
      result = NewRelic::Agent::Instrumentation::Roda::TransactionNamer.transaction_name({})

      assert_equal NewRelic::Agent::UNKNOWN_METRIC, result
      assert_logged(/NoMethodError.*Error encountered trying to identify Roda transaction name/)
```

---

Oh and finally:

- Add a comment to let future reviewers know that the use of `{}` as input was intended to raise a specific type of error.